H.Ws/H.W_4.py

```diff
@@ -0,0 +1,236 @@
+import numpy
+import pandas
+import statsmodels.api
+from plotly import express as px
+from plotly import figure_factory as ff
+from plotly import graph_objects as go
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.metrics import confusion_matrix
+from sklearn.model_selection import train_test_split
+
+titanic_df = pandas.read_csv(
+    "https://raw.githubusercontent.com/mwaskom/seaborn-data/master/titanic.csv"
+)  # .dropna().reset_index()
+
+print(titanic_df)
+print(titanic_df.dtypes)
+print(titanic_df.columns)
+predictors = ["pclass", "sex", "age", "sibsp", "embarked", "class"]
+response = ["alive"]
+
+x = titanic_df[predictors]
+y = titanic_df[response]
+resp_test = None
+pred_test = None
+
+# for response
+for val in response:
+    if y[val].dtype == "bool":
+        resp_test = "boolean"
+    else:
+        resp_test = "continuous"
+
+# for predictors
+for val in predictors:  # x.columns
+    if (
+        x[val].dtype == "string"
+        or x[val].dtype == "object"
+        or x[val].dtype == "bool"
+        or x[val].dtype == "category"
+    ):
+        pred_test = "categorical"
+    else:
+        pred_test = "continuous"
+
+
+# predictor - cont: Age
+# predictor - cat: sex, alone
+# response - cat: survived, alive
+
+# build code for all 4 senarios
+
+# 1_cat response - cat predictor
+def cat_response_cat_predictor():
+    conf_matrix = confusion_matrix(x[val], y[val])
+
+    fig_no_relationship = go.Figure(
+        data=go.Heatmap(z=conf_matrix, zmin=0, zmax=conf_matrix.max())
+    )
+    fig_no_relationship.update_layout(
+        title="Categorical Predictor by Categorical Response (with relationship)",
+        xaxis_title="Response",
+        yaxis_title="Predictor",
+    )
+    fig_no_relationship.show()
+    pass
```

Comment:
You keep on typing pass at the end of your functions, but that doesn't do anything. Pass is a null argument, when python interpreter see's it nothing happens. I believe this is going to cause you a lot of problems when you do function calls in your code.

Reply:
you should end your functions with "return" and if you want to return an object from your function you should use "return <object_name>